node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabase.java

```diff
@@ -94,14 +96,37 @@ public void create(Path path) {
 
     public Optional<byte[]> getData(Path path) { return getCache().getData(path); }
 
+    private static class CacheAndGeneration {
+        public CacheAndGeneration(CuratorDatabaseCache cache, long generation)
+        {
+            this.cache = cache;
+            this.generation = generation;
+        }
+        public boolean expired() {
+            return generation != cache.generation();
+        }
+        public CuratorDatabaseCache validCache() {
+            return !expired() ? cache : null;
```

Comment:
Woudn't it be better to throw an exception with some info instead of returning null, which will give a NullPointerException when the cache is used?

Reply:
You are probably right. For a man born is the sixties, thinking that nothing really new has happened in computer science since mid seventies, learning to program with fortran, c, pascal and the early days of C++, throwing an exception is a very hard thing to do. But I guess I have to , not embrace, but at least accept some changes.

Reply:
In C++ if is an error have no strategy for handling. Fx something that can not possibly happen. Getting a coredump is a lot better than throwing an exception that someone might catch and swallow. But coredumps are not an option in Java .....